sway-lib-core/src/raw_ptr.sw

```diff
@@ -0,0 +1,67 @@
+library raw_ptr;
+
+impl raw_ptr {
+    /// Returns `true` if the pointer is null.
+    pub fn is_null(self) -> bool {
+        let addr = asm(ptr: self) { ptr: u64 };
+        __eq(addr, 0)
+    }
+
+    /// Gets the address of the pointer.
+    pub fn addr(self) -> u64 {
+        asm(ptr: self) { ptr: u64 }
+    }
+
+    /// Calculates the offset from the pointer.
+    pub fn add(self, count: u64) -> raw_ptr {
+        let addr = asm(ptr: self) { ptr: u64 };
```

Comment:
Ugh too bad we can't use the methods of an `impl` block in other methods in the same `impl` block :(

This will get fixed soon though!